controllers/component_dependency_update_controller_test.go

```diff
@@ -192,27 +201,46 @@ var _ = Describe("Component nudge controller", func() {
 			}
 			pr.Status.CompletionTime = &metav1.Time{Time: time.Now()}
 			Expect(k8sClient.Status().Update(ctx, pr)).Should(BeNil())
+
 			Eventually(func() bool {
+				// check that no nudgeerror event was reported
 				events := v1.EventList{}
 				err := k8sClient.List(context.TODO(), &events)
 				Expect(err).ToNot(HaveOccurred())
-				op1nudged := false
-				op2nudged := false
+				failureCount := 0
 				for _, i := range events.Items {
-					if i.Reason == ComponentNudgedEventType && strings.Contains(i.Message, "quay.io.foo/bar:latest@sha256:12345") {
-						if strings.Contains(i.Message, "operator1") {
-							op1nudged = true
-						}
-						if strings.Contains(i.Message, "operator2") {
-							op2nudged = true
-						}
+					if i.Reason == ComponentNudgeFailedEventType {
+						failureCount++
+					}
+				}
+				// check that renovate config was created
```

Comment:
We have this code in at least two places. Do we need a separate function?

Reply:
ok created helper functions